thankyou_letters/models/partner_communication.py

```diff
@@ -33,6 +33,21 @@ class PartnerCommunication(models.Model):
         domain=[('type', '=', 'sentence')])
     success_sentence = fields.Text(related='success_sentence_id.body_text')
     add_success_story = fields.Boolean(related='config_id.add_success_story')
+    amount = fields.Float(string='Amount', compute='_compute_donation_amount',
+                          store=True)
+
+    @api.multi
+    def _compute_donation_amount(self):
+        for elem in self:
+            object_ids = map(int, elem.object_ids.split(','))
+            model = elem.config_id.model
+            amount_invoice = 0
+            for invoice_line_id in object_ids:
+                if model == 'account.invoice.line':
+                    invoice_line = self.env['account.invoice.line']\
+                        .search([('id', '=', invoice_line_id)])
+                    amount_invoice += invoice_line.price_subtotal
```

Comment:
You can directly sum all the amounts if you fetch all invoice_lines at once.
```suggestion
                    elem.amount = sum(invoice_lines.mapped('price_subtotal'))
```